packages/ic-asset-certification/tests/large_assets.rs

```diff
@@ -0,0 +1,193 @@
+use http::StatusCode;
+use ic_asset_certification::{Asset, AssetConfig, AssetEncoding, AssetRouter, ASSET_CHUNK_SIZE};
+use ic_certification_testing::{CertificateBuilder, CertificateData};
+use ic_http_certification::{
+    DefaultCelBuilder, DefaultResponseCertification, HeaderField, HttpRequest, HttpResponse,
+};
+use ic_response_verification::verify_request_response_pair;
+use ic_response_verification_test_utils::{create_canister_id, get_current_timestamp};
+use once_cell::sync::OnceCell;
+use rstest::*;
+
+mod common;
+use common::*;
+
+const ASSET_ONE_NAME: &str = "asset_one";
+const ASSET_ONE_SIZE: usize = ASSET_CHUNK_SIZE + 1;
+
+const MAX_CERT_TIME_OFFSET_NS: u128 = 300_000_000_000;
+const MIN_REQUESTED_VERIFICATION_VERSION: u8 = 2;
+
+#[rstest]
+fn test_large_assets(
+    asset_one_body: &'static Vec<u8>,
+    asset_one_chunk_one: &'static [u8],
+    asset_one_chunk_two: &'static [u8],
+) {
+    let current_time = get_current_timestamp();
+    let canister_id = create_canister_id("rdmx6-jaaaa-aaaaa-aaadq-cai");
+    let req_url = format!("/{}", ASSET_ONE_NAME);
+
+    let mut asset_router = AssetRouter::default();
+    let assets = [Asset::new(ASSET_ONE_NAME, asset_one_body)];
+    let asset_configs = [asset_config(ASSET_ONE_NAME.to_string(), vec![])];
+    asset_router.certify_assets(assets, asset_configs).unwrap();
+
+    let certified_data = asset_router.root_hash();
+    let CertificateData {
```

Comment:
This is one of the main differences between this test and the existing tests. We create a full certificate with a valid root key and use this to create a fully verifiable `IC-Certificate` header.